### PR TITLE
Keep beacon and Matchmaker status for reuploaded cases

### LINKED ISSUE #4432: Matchmaker submission data get erased on case rerun
If a case has an associated submission to Matchmaker, it has a key named "mme_submission"

The key is not preserved when the case is rerun/(reuploaded2)

An example is cust003's case cutegopher

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Fix #4432 + keep also beacon submission status when a case gets reuplaoded

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`

1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test on hasta:**:
1. Deploy on stage (hasta)
2. Go to a case and pin a variant
3. Create a matchmaker submission for the case
image
<hr>
4. Go to hasta and re-upload that case (scout --conf <path_2_stage_conf_file.yaml> load case <path_to_case_config.yml> --keep-actions --update
5. Submission should be maintained
6. Repeat step 4 while on main branch
7. Previous Matchmaker submission should not be visible and it will ask you if you want to submit it

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by TB
- [x] tests executed by CR / TB

### CODE REVIEW

**Comment by northwestwitch** [comment]
### Given a case with MME submission

image

### Re-uploaded (scout --demo load case scout/demo/643594.config.yaml --keep-actions --update) using this branch:

image

### Re-uploaded (scout --demo load case scout/demo/643594.config.yaml --keep-actions --update) using main branch:

image

File: scout/adapter/mongo/case.py [hunk 1]
```diff
@@ -852,6 +880,7 @@ def load_case(self, config_data, update=False, keep_actions=True):
             old_evaluated_variants = list(
                 self.evaluated_variants(case_obj["_id"], case_obj["owner"])
             )
+            self.update_case_data_sharing(old_case=existing_case, new_case=case_obj)
```
Comment (northwestwitch) [inline scout/adapter/mongo/case.py:883]: This is the only change in this function. I don't know what's going on with black

**Comment by northwestwitch** [comment]
I actually changed my mind about the functionality of this PR: since beacon and matchmaker submissions are not removed when a case is re-uploaded, even if the case is uploaded without the  --keep-actions  option, the connection to beacon and matchmaker data should be preserved. I'll fix it accordingly

**Comment by TereseBo** [comment]
On this branch, having added a Mme submission to a case: 
image

After uploading the case again (with "scout --demo load case --keep-actions --update scout/demo/643594.config.yaml" or  "scout --demo load case --update scout/demo/643594.config.yaml")

The MatchMaker submission is still visible:
image

When performed on main, the MatchMaker submission is not visible after re-upload of case:

image

**Review by TereseBo — APPROVED** [review]

**Pull request merged**